### ISSUE TITLE
Implementing multi-instance  and hierarchical multi-instance learning

### ISSUE THREAD

**Issue by pevnak** [apache/incubator-mxnet#3515]
Hello all,
I have written two papers about solving multi-instance learning problems with neural network formalism.
[Simple mil](https://arxiv.org/abs/1609.07257)
and an extension to two multi-instance problems one embedded into the other [multi=mil](https://www.researchgate.net/publication/309040375_Discriminative_Models_for_Multi-instance_Problems_with_Tree_Structure).
I have implemented everything in my library, but as you can expect it is loosing its breath. Would it be possible to implement such functionality in mxnet, such that it can run gpu?

Thank you very much for the answer.

Tomas

**Comment by piiswrong** [comment]
Do you mean you implemented it in some other framework and you want to move it to mxnet?
Or do you mean you did it in mxnet and you want to merge it into mxnet master branch?

We well come all contributions. Please open a pull request and we can help you merge it.

**Comment by pevnak** [comment]
It is implemented outside mxnet and I would like to move to it, if possible.

**Comment by piiswrong** [comment]
Ok. Then what do you mean by "I have written two papers about solving multi-instance learning problems with mxnet."?

Anyway we welcome the interest. Are you having any difficulties?

**Comment by pevnak** [comment]
Sorry, I have made a mistake and I have corrected that.
The problem I face is with implementation of aggregation operators like mean and max. I have all instances from all bags stored in one matrix. Then on basis of "meta" information I average all instances from each bag. This operation leads to matrix with the same number of features but with number of samples equal to the number of bags. That said, I cannot used mean over one dimension  of the tensor, each bag gas different number of instances. Does mxnet support this operator?
